server/routes/regenerate.test.ts

```diff
@@ -93,12 +120,11 @@ describe('regenerate route when rootAgentId is provided', () => {
     expect(mockedLogger.error).toBeCalledTimes(1);
     expect(result.source).toMatchInlineSnapshot(`
       Object {
-        "createdTimeMs": 0,
-        "interactions": Array [],
-        "messages": Array [],
+        "interactions": Array [
+          undefined,
```

Comment:
Shall we remove `undefined` in the interactions array?

Reply:
Sure and it need a better error handling when return interactions and messages. Done